packages/components/src/internal/components/notifications/actions.ts

```diff
@@ -44,10 +44,145 @@ export function createNotification(creatable: NotificationCreatable) {
  * Used to notify the server that the trial banner has been dismissed
  */
 export function setTrialBannerDismissSessionKey(): Promise<any> {
-    return new Promise((resolve, reject) => {
+    return new Promise(() => {
         Ajax.request({
             url: buildURL('core', 'dismissWarnings.api'),
             method: 'POST',
         });
     });
 }
+
+export function getServerNotifications(typeLabels?: string[], maxRows?: number): Promise<ServerActivity> {
+    return new Promise((resolve, reject) => {
+        Ajax.request({
+            url: ActionURL.buildURL('notification', 'getUserNotifications.api'),
+            method: 'GET',
+            params: { container: getServerContext().container.id, typeLabels, maxRows },
+            success: Utils.getCallbackWrapper(response => {
+                if (response.success) {
+                    const notifications = response.notifications.map(
+                        notification => new ServerActivityData(notification)
+                    );
+                    resolve({
+                        data: notifications,
+                        totalRows: response.totalRows,
+                        unreadCount: response.unreadCount,
+                        inProgressCount: 0,
+                    });
+                } else {
+                    console.error(response);
+                    reject('There was a problem retrieving your notification data.');
+                }
+            }),
+            failure: Utils.getCallbackWrapper(response => {
+                console.error(response);
+                reject(resolveErrorMessage(response));
+            }),
+        });
+    });
+}
+
+export function getRunningPipelineJobStatuses(): Promise<ServerActivity> {
```

Comment:
I do wonder, if later we want to enable notifications for inventory (independent of LKSM), what would its notifications look like.

Reply:
We'll surely make modifications here once we do that. Even when we add notifications for workflow this will be different, but since we don't know exactly how we want to support this I'm leaving it as is for now.